### PR TITLE
Fixes #165 - npm run remotetest completes cleanly

### LINKED ISSUE #165: remotetests are not completing successfully
npm run remotetest does not complete successfully.

### PR DESCRIPTION
**Pull request by kurtkanaskie**
While fixing remotetest.js, identified and fixed additional issues #158 #160 #161  and #163 

Unfortunately, there are a lot of formatting changes to remotetest.js

### CODE REVIEW

**Comment by kurtkanaskie** [comment]
The gist of these fixes are:
- remotetest.js was modified to flatten the describe / it tests which makes it much easier to understand the tests and their flows. This caused lots of edits, sorry.
- remotetest.js was enhanced to remove any test artifacts (e.g. proxies) and now leaves the org as it was before testing.
- options.js and defaults.js where modified to identify "default" options so the "command specific" options can be highlighted in help
- Error handling and tests were improved to detect missing options.
- The following issues where also fixed in this merge: 158, 160, 161, 163.

**Comment by kurtkanaskie** [comment]
Test for Issue #159 
$ mocha -R spec remotetests --grep "SharedFlows and FlowHooks" 
  SharedFlows and FlowHooks
    ✓ Deploy SharedFlow (3691ms)
    ✓ listSharedFlowDeployments (442ms)
    ✓ fetchSharedFlow (194ms)
    ✓ getPreviousSharedFlow (144ms)
    ✓ attachFlowHook (493ms)
    ✓ detachFlowHook (238ms)
    ✓ re-attachFlowHook (584ms)
    ✓ undeploySharedFlow (1513ms)
    ✓ deleteSharedFlow (702ms)
  9 passing (8s)

**Comment by kurtkanaskie** [comment]
Test for Issue #162 
$ mocha -R spec remotetests --grep "Target Servers"
  Target Servers
    ✓ Create Target Server (512ms)
    ✓ List Target Servers (173ms)
    ✓ Get Target Server (165ms)
    ✓ Delete Target Server (372ms)
  4 passing (1s)

**Comment by kurtkanaskie** [comment]
Test for Issue #167 
$ mocha -R spec remotetests --grep "User Roles and Permissions"
  User Roles and Permissions
    ✓ Create Role (329ms)
    ✓ Get Role (228ms)
    ✓ List Roles (252ms)
    ✓ Set Role Permissions (183ms)
    ✓ Get Role Permissions (175ms)
    ✓ Assign User to Role (247ms)
    ✓ Verify User in Role (191ms)
    ✓ List Users in a Role (253ms)
    ✓ Verify access allowed (156ms)
    ✓ Remove User from Role (166ms)
    ✓ Delete Role (183ms)
  11 passing (2s)

**Comment by kurtkanaskie** [comment]
All done!
$ npm test
> apigeetool@0.12.0 test /Users/kurtkanaskie/work/apigee/apigeetool-prs/apigeetool-node-PRS
> jasmine test/testoptions.js
Started
..............
14 specs, 0 failures
Finished in 0.029 seconds

$ npm run remotetest (NOTE: hosted target fails due to current omg/16078)
> apigeetool@0.12.0 remotetest /Users/kurtkanaskie/work/apigee/apigeetool-prs/apigeetool-node-PRS
> mocha -R spec remotetests
  Remote Tests
    ✓ Deploy Apigee Proxy (15110ms)
    ✓ Create Product (320ms)
    ✓ Create Private Product (270ms)
    ✓ Create Developer (294ms)
    ✓ Create App (2038ms)
    ✓ Delete App (732ms)
    ✓ Delete Developer (556ms)
    ✓ Delete API Product (435ms)
    ✓ Delete API private Product (428ms)
    ✓ Deploy Apigee Proxy (75616ms)
    ✓ Verify deployed URI (120ms)
    ✓ List Deployments by app (1330ms)
    ✓ List Deployments by environment (5367ms)
    ✓ Undeploy Apigee Proxy With Revision (1461ms)
    ✓ Deploy Apigee Proxy with base path (15520ms)
    ✓ Verify deployed URI (100ms)
    ✓ Deploy Apigee Proxy with base path and run NPM remotely (76821ms)
    ✓ Verify deployed URI (115ms)
    ✓ Undeploy Apigee Proxy With Revision (988ms)
    ✓ Fetch proxy (1756ms)
    ✓ Delete proxy (667ms)

  Node.js Apps
    ✓ Deploy Node.js App (14833ms)
    ✓ Verify deployed URI (99ms)
    ✓ Check logs from deployed URI (268ms)
    ✓ Deploy Node.js App and run NPM remotely (89099ms)
    ✓ Verify deployed URI (123ms)
    ✓ List Deployments by app (1373ms)
    ✓ Undeploy Node.js App Without Revision (1441ms)
    ✓ Delete node proxy (429ms)

  Hosted Target
    1) Deploy Hosted Targets App
    2) List Deployments by app
    3) Verify deployed URI
    ✓ Check build logs from deployed URI (520ms)
    4) Check runtime logs from deployed URI
    5) Undeploy Hosted Targets App Without Revision
    ✓ Delete hosted target proxy (1752ms)

  Caches
    ✓ Create an Cache Resource (749ms)
    ✓ Delete Cache Resource (565ms)

  Target Servers
    ✓ Create Target Server (321ms)
    ✓ List Target Servers (189ms)
    ✓ Get Target Server (168ms)
    ✓ Delete Target Server (562ms)

  KVM
    ✓ Create KVM (373ms)
    ✓ Create Encrypted KVM (611ms)
    ✓ Delete Encrypted KVM (374ms)
    ✓ Add Entry to KVM (1097ms)
    ✓ Get KVM Entry (408ms)
    ✓ Get KVM Map (585ms)
    ✓ Delete KVM Entry (642ms)
    ✓ Delete KVM (351ms)

  SharedFlows and FlowHooks
    ✓ Deploy SharedFlow (2587ms)
    ✓ listSharedFlowDeployments (234ms)
    ✓ fetchSharedFlow (214ms)
    ✓ getPreviousSharedFlow (179ms)
    ✓ attachFlowHook (762ms)
    ✓ detachFlowHook (227ms)
    ✓ re-attachFlowHook (562ms)
    ✓ undeploySharedFlow (1228ms)
    ✓ deleteSharedFlow (353ms)

  User Roles and Permissions
    ✓ Create Role (154ms)
    ✓ Get Role (135ms)
    ✓ List Roles (229ms)
    ✓ Set Role Permissions (234ms)
    ✓ Get Role Permissions (179ms)
    ✓ Assign User to Role (351ms)
    ✓ Verify User in Role (175ms)
    ✓ List Users in a Role (152ms)
    ✓ Verify access allowed (190ms)
    ✓ Remove User from Role (166ms)
    ✓ Delete Role (262ms)

  65 passing (7m)
  5 failing

File: README.md [hunk 1]
```diff
@@ -76,9 +82,6 @@ trusted TLS certificate.
 (optional) Limit for the maximum number of operations performed concurrently.
 Currently this only affects file uploads in the `deploynodeapp` command. Defaults to 4.
 
-`--json  -j`
```
Comment (whitlockjc) [inline README.md]: Why are we removing this?

Comment (kurtkanaskie) [reply README.md]: We're not, my bad, I wasn't sure what it was for, but now I do.
I'll add it back.

**Pull request merged**